src/gui/qgsfilewidget.cpp

```diff
@@ -359,16 +388,30 @@ void QgsFileDropEdit::setFilters( const QString &filters )
 
 QString QgsFileDropEdit::acceptableFilePath( QDropEvent *event ) const
 {
-  QString path;
+  QStringList paths;
   if ( event->mimeData()->hasUrls() )
   {
-    QFileInfo file( event->mimeData()->urls().first().toLocalFile() );
-    if ( ( mStorageMode == QgsFileWidget::GetFile && file.isFile() &&
-           ( mAcceptableExtensions.isEmpty() || mAcceptableExtensions.contains( file.suffix(), Qt::CaseInsensitive ) ) )
-         || ( mStorageMode == QgsFileWidget::GetDirectory && file.isDir() ) )
-      path = file.filePath();
+    for ( const auto url : event->mimeData()->urls() )
```

Comment:
To be honest I knew it caused a full copy, but in this case I don't think we have any noticeable overhead. BTW, a `Q_FOREACH` would be acceptable?